### DISCUSSION TITLE
Does Vercel support .mjs files?

### DISCUSSION THREAD

**Discussion by rootedsoftware** [vercel/vercel#5864] [Help] [answered]
I was having an issue with the build process thinking one of my files in the /api folder was a common js file, so I renamed it to .mjs, but now it just doesn't show up after the build completed. I think the answer to my question must be no, or maybe there is another way to force it to build the file as an ES module. I know I could put type: "module" in the package.json file, but there are other files that I don't want to change.

**Comment by styfle** [comment, marked as answer]
We don't currently support `.mjs` or `type: "module"` for Node.js Serverless Functions but we will soon, thanks!

**Reply by styfle** [reply]
Hi @c316 

We have experimental support for ESM on the canary channel.

You can try it out now by adding the environment variable `FORCE_RUNTIME_TAG=canary` to your project and deploying again.
